### ISSUE TITLE
Implement Non-Blocking Async Threading

### ISSUE THREAD

**Issue by wilson97** [deepgram/deepgram-python-sdk#301]
Hi, I recently upgraded my python deepgram client from 2.11 to 3.1. I followed the new 3.1 example here: https://github.com/deepgram/deepgram-python-sdk/blob/a6fdb9f18b717961ac8e26f30ebbc57dd2772939/examples/streaming/async_http/main.py

Unfortunately there is a change in the async client behavior undocumented from 2.11 to 3.1. In 2.11, whenever a new message comes, you call asyncio.create_task() (https://github.com/deepgram/deepgram-python-sdk/blob/2.11.0/deepgram/transcription.py#L261), which spawns a new handler function that runs in the background.

However, in 3.1, you await the coroutine, so it doesn't run in the background anymore (https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/live/v1/async_client.py#L114).

So say I have five different handlers all registered that just execute asyncio.sleep(1). In 2.11, these five handlers would run in parallel. In 3.1, these five different handlers would block each other, run sequentially, and take 5 seconds total. 

Am I misunderstanding the code here? Why was this change in behavior made? Thanks.

**Comment by dvonthenen** [comment]
Hi @wilson97 

Thanks for the feedback. This is something we can take a look at. There was some initial work done that we never got the time to revisit because of time constraints. Most of this affects the async listen client. I appreciate your input.

**Comment by dvonthenen** [comment]
The workaround for this is to create task the in your threads to not block on that.

**Comment by dvonthenen** [comment]
Hi @wilson97 

If you had the time, can you take a look at this PR to see if it addresses your issue? https://github.com/deepgram/deepgram-python-sdk/pull/309

**Comment by dvonthenen** [comment]
This along with other Async Client improvements have been merged into `main`.
